sdk/tests/test_openstack.py

```diff
@@ -87,73 +88,354 @@ def test_list_vms(mock_openstack):
     mock_compute.servers.assert_called_once()
 
 
-@pytest.mark.skip(reason="Refactoring: will update in follow-up PR")
 @mock.patch("openstack.connection.Connection")
-def test_create_vm(mock_openstack):
-    """Test successful creation of a VM."""
+def test_list_vms_with_status_filter(mock_openstack):
+    """Test listing VMs with status filter."""
     mock_compute = mock.MagicMock()
     server = Mock(
         id="42",
         flavor={"original_name": "rhel-10"},
-        availability_zone="us",
         addresses={
             "private": [
                 {"OS-EXT-IPS:type": "fixed", "addr": "10.123.50.11", "version": "ipv4"}
             ]
         },
         key_name="my_key",
-        status="ACTIVE",
+        status="SHUTOFF",
     )
     type(server).name = PropertyMock(return_value="server")
 
-    mock_compute.create_server.return_value = server
+    mock_compute.servers.return_value = [server]
     mock_openstack.return_value.compute = mock_compute
 
     openstack_helper = OpenStackHelper()
-    result = openstack_helper.create_vm(
-        ["server", "image_rhel-10.iso", "rhel-10", "private"]
-    )
+    result = openstack_helper.list_servers({"status": "SHUTOFF"})
 
     assert result["count"] == 1
+    instances = result["instances"]
+    assert len(instances) == 1
+    assert instances[0].get("status") == "SHUTOFF"
+
+    mock_openstack.assert_called_once()
+    mock_compute.servers.assert_called_once_with(status="SHUTOFF")
+
+
+@mock.patch("openstack.connection.Connection")
+def test_create_servers_success(mock_openstack):
+    """Test successful creation of a VM."""
+    mock_compute = mock.MagicMock()
+    mock_flavor = Mock(id="flavor-id-123")
+    mock_flavor.name = "rhel-10"
+    mock_image = Mock(id="image-id-456")
+
+    # Mock the server object that gets created
+    mock_server = Mock(
+        id="server-id-789",
+        status="ACTIVE",
+        addresses={
+            "private": [
+                {"OS-EXT-IPS:type": "fixed", "addr": "10.123.50.11", "version": "ipv4"}
+            ]
+        },
+    )
+    mock_server.name = "test-server"
+
+    # Mock the keypairs properly
+    mock_keypair = Mock()
+    mock_keypair.name = "test-key"
+    mock_compute.keypairs.return_value = [mock_keypair]
+
+    # Mock flavor and image finding
+    mock_compute.find_flavor.return_value = mock_flavor
+    mock_compute.find_image.return_value = mock_image
+
+    # Mock server creation and waiting
+    mock_compute.create_server.return_value = mock_server
+    mock_compute.wait_for_server.return_value = mock_server
```

Comment:
```suggestion
    mock_server = Mock(
        id="server-id-789",
        status="ACTIVE",
        addresses={
            "private": [
                {"OS-EXT-IPS:type": "fixed", "addr": "10.123.50.11", "version": "ipv4"}
            ]
        },
    )
    mock_server.name = "test-server"

    mock_keypair = Mock()
    mock_keypair.name = "test-key"
    mock_compute.keypairs.return_value = [mock_keypair]

    mock_compute.find_flavor.return_value = mock_flavor
    mock_compute.find_image.return_value = mock_image

    mock_compute.create_server.return_value = mock_server
    mock_compute.wait_for_server.return_value = mock_server
```

I think these comments are redundant because mock is a mock and the actions are explained by the methods itself.